### PR TITLE
[ch907] sheets: adding support for uploading to Google Sheets

### PR DESCRIPTION
**Pull request by damjanovl**

### CODE REVIEW

**Comment by blakeburch** [comment]
So there's no confusion on Syntax:
File = Collective Google Sheet
Tab = Individual Sheet on a Google Sheet

1. Per our discussion in Slack, need to rename variables to be more abstract (so they can be easily transitioned between spreadsheet vendors). I'm leaning towards --destination-file-name and --tab-name
2. For test 15, we want the script to fail when the File does not exist. I'm assuming because it doesn't fail, it's creating a new File. For something like Google sheets, where visualizing and collaboration is key, we don't want "invisible sheets" to exist in the service account.
3. Test 17 and 18 should still work. If a starting spot is specified that is outside standard or current grid limits, it should attempt to expand the grid to the specified starting value. Fortunately, if the starting cell already exists within the grid, Google Sheets will automatically expand the grid for the values being added in.
```
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://sheets.googleapis.com/v4/spreadsheets/IDHERE/values:batchUpdate?alt=json returned "Invalid data[0]: Range ('Outside Bounds'!AA1:ZZZ5000000) exceeds grid limits. Max rows: 1000, max columns: 26">
```

4. For Test 22, an error is being presented even though it works correctly. Something about the spaces in the tab name seems to be throwing it off.
```
<HttpError 400 when requesting https://sheets.googleapis.com/v4/spreadsheets/IDHERE:batchUpdate?alt=json returned "Invalid requests[0].addSheet: A sheet with the name "Sheet with spaces" already exists. Please enter another name.">
```
5. CSVs aren't being properly imported into Google Sheets. Quoted fields are still showing and some of the rows get messed up. We want the final behavior to reflect the same output that you would get by using the import button in the UI. See the video in slack.
6. For test 2 and 3 with larger file sizes, I kept getting an issue of `line contains NUL`. When using the importer built into Google Sheets, it had no problem.

**Comment by blakeburch** [comment]
@damjanovl Getting closer! Files uploaded in the correct, expected format and no new issues were found.

Issue no2 still exists.
`python upload_file.py --source-file-name 'small.csv' --destination-file-name 'does-not-exist' --tab-name 'Sheet1' --starting-cell 'A1' --service-account "creds.json"`

Issue no3 still exists.

Issue no4 still exists

File: sheets/google_sheets/upload_file.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    Checks if the workbook exists within the spreadsheet.
+    """
+    try:
+        spreadsheet = service.spreadsheets().get(spreadsheetId=spreadsheet_id).execute()
+        sheets = spreadsheet['sheets']
+        exists = [True for sheet in sheets if sheet['properties']['title'] == tab_name]
+        return True if exists else False
+    except Exception as e:
+        print(f'Failed to check workbook {tab_name} for spreadsheet ' \
+                f'{spreadsheet_id}')
```
Comment (blakeburch) [inline sheets/google_sheets/upload_file.py:92]: These dynamic variables should be flipped and contain Google Sheet syntax. I got a connection error that confused me here.

'Failed to check spreadsheet {spreadsheet_id} for a sheet named {tab_name}`

**Comment by blakeburch** [comment]
Forgot to add.

7) Needs to have support for the --drive flag, uploading data to Files located within a Shared Drive.

**Comment by blakeburch** [comment]
Perfecto. This one is good to go.

File: sheets/google_sheets/upload_file.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+import glob
+
+from googleapiclient.discovery import build
+from googleapiclient.http import MediaFileUpload
+from google.oauth2 import service_account
+
+SCOPES = ['https://spreadsheets.google.com/feeds',
+          'https://www.googleapis.com/auth/drive']
+socket.setdefaulttimeout(600)
+#SCOPES = ['https://www.googleapis.com/auth/spreadsheets']
```
Comment (ericelsken) [inline sheets/google_sheets/upload_file.py:17]: Remove this line.

**Pull request merged**